backend/modules/simulation/service.py

```
@@ -132,6 +132,16 @@ async def process_chat_message(
             )
             
             if progression_result.get('simulation_complete'):
+                # Clean up sandbox if one was created
+                if user_progress.sandbox_id:
+                    try:
+                        from common.services.sandbox_service import sandbox_service
+                        await sandbox_service.delete_sandbox(user_progress.sandbox_id)
+                        logger.info(f"[SERVICE] Cleaned up sandbox {user_progress.sandbox_id}")
+                    except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)